google_sso/README.md

```diff
@@ -0,0 +1,307 @@
+
+Take a note of the VM's external IP address, you will need it in the subsequent steps.
+We will refer to is as VM_EXTERNAL_IP in the rest of the guide.
+
+### Configuring OAuth
+
+In order to use HAS with Google Identity, you will need to configure an OAuth 2.0 client.
+In the Google Cloud Console, navigate to API & Services and select Credentials.
+Click on Create Credentials, select "OAuth client ID" and pick "Web application"
+as the Application type:
+
+![Create Credentials](create_credentials.png)
+
+You can name the credentials "HAS Client" (the name does not matter).
+
+Note that you will need to provide the authorized Authorized redirect URIs.
+Since we are not using a domain name but an IP address, we would need to use a special domain,
+xip.io, that does the redirect. You can see the following post for more details:
+https://stackoverflow.com/questions/14238665/can-a-public-ip-address-be-used-as-google-oauth-redirect-uri
+
+The two URIs you'll need to list are:
+
+-  https://VM_EXTERNAL_IP.xip.io:3000
+-  https://VM_EXTERNAL_IP.xip.io:3000/oidc/callback
+
+If this the first time you are creating OAuth credentials, you will also need to configure
+an OAuth consent screen.
+
+Once you have created the credentials, take note of the Client ID and the Client Secret.
+These values will be needed when configuring HAS.
+
+### Configuring the firewall
+
+We will most likely want to access the Perforce VM from outside of the default VPC network.
+For the purpose of this exercise, we will open up the VPC firewall to allow unrestricted
+access to our VMs on ports 1666 (Helix Core) and 3000 (HAS).
+
+**Please consult a security or network engineer when adapting this to your environment.**
+
+To open up all VMs on your GCP network to incoming traffic on ports 1666 and 3000, follow these steps:
+
+-  In Google Cloud Console, navigate to VPC Network / Firewall
+-  Click on "Create Firewall Rule"
+
+![Create Firewall](create_firewall.png)
+
+You can give your rule any name (e.g. "helix-and-has"). Select "All instances in the network"
+as targets, choose 0.0.0.0/0 as source IP ranges and specify TCP ports 1666,3000.
+
+Note: While this rule is fine for testing, consider restricting them to specific VMs by using
+target tags.
+
+### Installing the software
+
+You will need to ssh into the VM to install Helix Core and HAS.
+This can be done by clicking the "SSH" button in the VM instance list in the Google Cloud Console.
+Alternatively, you can use the gcloud command:
+
+```
+gcloud compute ssh perforce
+```
+
+#### Installing and configuring Helix Core
+
+The full guide on installing Helix Core is available here:
+https://www.perforce.com/manuals/p4sag/Content/P4SAG/install.linux.packages.install.html
+
+In a nutshell:
+
+```
+wget -qO - https://package.perforce.com/perforce.pubkey | sudo apt-key add -
+sudo sh -c "echo 'deb http://package.perforce.com/apt/ubuntu xenial release' > /etc/apt/sources.list.d/perforce.list"
```

Comment:
Better to use sudo tee then sudo sh.
`echo 'deb http://package.perforce.com/apt/ubuntu xenial release' | sudo tee /etc/apt/sources.list.d/perforce.list`